pkg/collector/corechecks/snmp/config.go

```diff
@@ -0,0 +1,271 @@
+	MetricTags       []metricTagConfig `yaml:"metric_tags"`
+	Profile          string            `yaml:"profile"`
+	UseGlobalMetrics bool              `yaml:"use_global_metrics"`
+}
+
+type snmpConfig struct {
+	ipAddress         string
+	port              uint16
+	communityString   string
+	snmpVersion       string
+	timeout           int
+	retries           int
+	user              string
+	authProtocol      string
+	authKey           string
+	privProtocol      string
+	privKey           string
+	contextName       string
+	oidConfig         oidConfig
+	metrics           []metricsConfig
+	metricTags        []metricTagConfig
+	oidBatchSize      int
+	profiles          profileDefinitionMap
+	profileTags       []string
+	uptimeMetricAdded bool
+}
+
+func (c *snmpConfig) refreshWithProfile(profile string) error {
+	if _, ok := c.profiles[profile]; !ok {
+		return fmt.Errorf("unknown profile `%s`", profile)
+	}
+	log.Debugf("Refreshing with profile `%s` with content: %#v", profile, c.profiles[profile])
+	tags := []string{"snmp_profile:" + profile}
+	definition := c.profiles[profile]
+
+	c.metrics = append(c.metrics, definition.Metrics...)
+	c.metricTags = append(c.metricTags, definition.MetricTags...)
+	c.oidConfig.scalarOids = append(c.oidConfig.scalarOids, parseScalarOids(definition.Metrics, definition.MetricTags)...)
+	c.oidConfig.columnOids = append(c.oidConfig.columnOids, parseColumnOids(definition.Metrics)...)
+
+	if definition.Device.Vendor != "" {
+		tags = append(tags, "device_vendor:"+definition.Device.Vendor)
+	}
+	c.profileTags = tags
+	return nil
+}
+
+func (c *snmpConfig) addUptimeMetric() {
+	if c.uptimeMetricAdded {
+		return
+	}
+	metricConfig := getUptimeMetricConfig()
+	c.metrics = append(c.metrics, metricConfig)
+	c.oidConfig.scalarOids = append(c.oidConfig.scalarOids, metricConfig.Symbol.OID)
+	c.uptimeMetricAdded = true
+}
+
+func (c *snmpConfig) getStaticTags() []string {
+	tags := []string{"snmp_device:" + c.ipAddress}
+	return tags
+}
+
+// toString used for logging, it will hide sensitive information
+func (c *snmpConfig) toString() string {
+	configCopy := *c
+	configCopy.communityString = "***"
+	configCopy.authKey = "***"
+	configCopy.privKey = "***"
+	return fmt.Sprintf("%#v", configCopy)
+}
+
+func buildConfig(rawInstance integration.Data, rawInitConfig integration.Data) (snmpConfig, error) {
+	instance := snmpInstanceConfig{}
+	initConfig := snmpInitConfig{}
+
+	// Set defaults before unmarshalling
+	instance.UseGlobalMetrics = true
+
+	err := yaml.Unmarshal(rawInitConfig, &initConfig)
+	if err != nil {
+		return snmpConfig{}, err
+	}
+
+	err = yaml.Unmarshal(rawInstance, &instance)
+	if err != nil {
+		return snmpConfig{}, err
+	}
+
+	c := snmpConfig{}
+
+	c.snmpVersion = instance.SnmpVersion
+	c.ipAddress = instance.IPAddress
+	c.port = uint16(instance.Port)
+
+	if c.port == 0 {
+		c.port = defaultPort
+	}
+
+	if instance.Retries == 0 {
+		c.retries = defaultRetries
+	} else {
+		c.retries = int(instance.Retries)
```

Comment:
I needed to do a [type conversion](https://golang.org/ref/spec#Conversions) here because `instance.Retries` have `Number` type (which is an alias for `int`) otherwise the compiler will complain.